src/loaders/sass.ts

```diff
@@ -1,20 +1,29 @@
+import { dirname, basename } from "pathe";
 import type { Loader, LoaderResult } from "../loader";
 
-export const sassLoader: Loader = async (input, { options }) => {
+export const sassLoader: Loader = async (input) => {
   if (![".sass", ".scss"].includes(input.extension)) {
     return;
   }
-
+  if (basename(input.srcPath).startsWith('_')) {
+    // sass files starting with "_" are always considered partials
+    // and should not be compiled to standalone CSS
+    return [{
+      contents: '',
+      path: input.path,
+      skip: true
+    }]
+  }
   const compileString = await import("sass").then(r => r.compileString || r.default.compileString);
 
   const output: LoaderResult = [];
 
   const contents = await input.getContents();
 
   output.push({
-    contents: compileString(contents).css,
+    contents: compileString(contents, { loadPaths: [dirname(input.srcPath)]}).css,
```

Comment:
`loadPaths` is required to enable relative imports (https://github.com/sass/sass/issues/3366#issuecomment-1203193294)